frontend/src/app/corpus-definitions/corpus-definition.service.ts

```diff
@@ -123,6 +144,47 @@ export class CorpusDefinitionService implements OnDestroy {
         return field as APICorpusDefinitionField;
     }
 
+    public updateDocumentationPage(
+        page: EditablePage,
+        data: CorpusDocumentationPage[]
+    ) {
+        const stored = data.find(p => p.type == page.category.title);
+        if (stored) {
+            page.id = stored.id;
+            page.content = stored.content_template;
+        } else {
+            page.id = undefined;
+            page.content = '';
+        }
+    }
+
+    public saveDocumentationPages(pages: EditablePage[]): Observable<any[]> {
+        return forkJoin(
+            pages.map(page => this.saveDocumentationPage(page))
+        ).pipe(
+            tap(() => this.documentationUpdated$.next())
+        );
+    }
+
+    private saveDocumentationPage(
+        page: EditablePage,
+    ): Observable<any> {
+        const data: CorpusDocumentationPageSubmitData = {
+            content_template: page.content,
+            type: page.category.title,
+            corpus: page.corpusName,
+        };
+        if (page.id && page.content.length) {
+            return this.apiService.updateCorpusDocumentationPage(page.id, data)
+        } else if (page.content.length) {
+            return this.apiService.createCorpusDocumentationPage(data);
+        } else if (page.id) {
+            return this.apiService.deleteCorpusDocumentationPage(page.id);
+        } else {
+            return of(undefined);
```

Comment:
Was going to recommend a `switch...case` for this but after reading the documentation, let's not.